lib/network/rest/client.js

```diff
@@ -0,0 +1,267 @@
+            });
+        }).on('finish', () => {
+            log.debug('finished sending PUT data to the REST server', {
+                component: 'RESTClient',
+                method: 'put',
+                contentLength: size,
+            });
+        }).on('error', callback);
+
+        // disable nagle algorithm
+        request.setNoDelay(true);
+
+        stream.pipe(request);
+        stream.on('error', err => {
+            log.error('error from readable stream', {
+                error: err,
+                method: 'put',
+                component: 'RESTClient',
+            });
+            request.end();
+        });
+    }
+
+    /**
+     * send a GET request to the REST server
+     * @param {String} key - The key associated to the value
+     * @param { Number [] | Undefined} range - range (if any) a
+     *   [start, end] inclusive range specification, as defined in
+     *   HTTP/1.1 RFC.
+     * @param {String} reqUids - The serialized request ids
+     * @param {RESTClient~getCallback} callback - callback
+     * @returns {undefined}
+     */
+    get(key, range, reqUids, callback) {
+        const log = this.createLogger(reqUids);
+        const headers = {};
+        setRequestUids(headers, reqUids);
+        if (range) {
+            setRange(headers, range);
+        }
+        const reqHeader = this.createRequest('GET', headers, key);
+        log.debug('about to send GET request', {
+            hostname: reqHeader.hostname,
+            port: reqHeader.port,
+            path: reqHeader.path,
+            headers: reqHeader.headers });
+        const request = http.request(reqHeader, response => {
+            response.once('readable', () => {
+                if (response.statusCode !== 200 &&
+                    response.statusCode !== 206) {
+                    return callback(makeErrorFromHTTPResponse(response));
+                }
+                return callback(null, response);
+            });
+        }).on('error', callback);
+
+        // disable nagle algorithm
+        request.setNoDelay(true);
+        request.end();
+    }
+
+    /**
+     * send a DELETE request to the REST server
+     * @param {String} key - The key associated to the values
+     * @param {String} reqUids - The serialized request ids
+     * @param {RESTClient~deleteCallback} callback - callback
+     * @returns {undefined}
+     */
+    delete(key, reqUids, callback) {
+        const log = this.createLogger(reqUids);
+        const headers = {};
+        setRequestUids(headers, reqUids);
+
+        const reqHeader = this.createRequest('DELETE', headers, key);
+        log.debug('about to send DELETE request', {
+            hostname: reqHeader.hostname,
+            port: reqHeader.port,
+            path: reqHeader.path,
+            headers: reqHeader.headers });
+        const request = http.request(reqHeader, response => {
+            response.once('readable', () => {
+                if (response.statusCode !== 200) {
```

Comment:
Apparently either a 200 or 204 is ok for a DELETE status code. So in our case it can indeed make more sense to return 204 since we don't return any body.

I like this chart ^^:
https://i.stack.imgur.com/whhD1.png